webtau-groovy/src/main/groovy/com/twosigma/webtau/cli/interactive/WebTauCliInteractive.groovy

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2018 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.cli.interactive
+
+import com.twosigma.webtau.WebTauGroovyDsl
+import com.twosigma.webtau.cfg.GroovyRunner
+import com.twosigma.webtau.console.ConsoleOutputs
+import com.twosigma.webtau.console.ansi.AutoResetAnsiString
+import com.twosigma.webtau.console.ansi.Color
+import com.twosigma.webtau.reporter.ConsoleStepReporter
+import com.twosigma.webtau.reporter.IntegrationTestsMessageBuilder
+import com.twosigma.webtau.reporter.StepReporters
+import com.twosigma.webtau.reporter.stacktrace.StackTraceUtils
+import com.twosigma.webtau.runner.standalone.StandaloneTest
+import com.twosigma.webtau.runner.standalone.StandaloneTestListeners
+import com.twosigma.webtau.runner.standalone.StandaloneTestRunner
+import com.twosigma.webtau.runner.standalone.report.StandardConsoleTestListener
+
+import java.nio.file.Paths
+
+import static com.twosigma.webtau.cfg.WebTauConfig.getCfg
+
+class WebTauCliInteractive {
+    enum State {
+        TestSelection,
+        ScenarioSelection,
+        ScenarioActionSelection,
+        AfterRun,
+        Done
+    }
+
+    private StandaloneTestRunner runner
+    private LinkedHashSet<String> testFilePaths
+    private Map<String, List<StandaloneTest>> testByPath = [:]
+    private BufferedReader inReader
+
+    private State state
+
+    private int selectedPathIdx = -1
+    private int selectedScenarioIdx = -1
+
+    WebTauCliInteractive(StandaloneTestRunner runner) {
+        this.runner = runner
+        this.testFilePaths = new LinkedHashSet<>(runner.tests.reportTestEntry.filePath*.toString())
```

Comment:
Pardon my groovy ignorance, what does the * do/mean?

Reply:
call a method on each element of a list